src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> elementToEdit = getNode(index);
+        T oldElement = elementToEdit.item;
+        elementToEdit.item = value;
+        return oldElement;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> elementToRemove = getNode(index);
+        unlink(elementToRemove);
+        return elementToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentValue = head;
+        for (int i = 0; i < size; i++) {
+            if (isElementsEqual(currentValue.item, object)) {
+                unlink(currentValue);
+                return true;
+            }
+            currentValue = currentValue.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getNode(int index) {
+        indexCheckForGet(index);
+        Node<T> currentValue;
+        if ((size / 2) > index) {
+            currentValue = head;
+            for (int i = 0; i < index; i++) {
+                currentValue = currentValue.next;
+            }
+        } else {
+            currentValue = tail;
+            for (int i = size; i > (index + 1); i--) {
+                currentValue = currentValue.prev;
+            }
+        }
+        return currentValue;
+    }
+
+    private void unlink(Node<T> node) {
+        size--;
+        if (node.next == null) {
+            if (node.prev != null) {
+                node.prev.next = null;
+            }
+            tail = node.prev;
+        } else if (node.prev == null) {
+            node.next.prev = null;
+            head = node.next;
+        } else {
+            node.next.prev = node.prev;
+            node.prev.next = node.next;
+        }
+
+    }
+
+    private boolean isElementsEqual(T element1, T element2) {
+        return element1 == element2
+                || (element1 != null && element1.equals(element2));
+    }
+
+    private void indexCheckForGet(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("This index " + index + " out of scope");
+        }
+    }
+
+    private void indexCheckForAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("This index " + index + " out of scope");
+        }
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.next = next;
+            this.item = element;
```

Comment:
use this. only if you have the same name variable. In other cases it is superfluous. you can rename element to item.